src/components/App.js

```diff
@@ -1,28 +1,164 @@
-import React, { useState, useEffect } from 'react';
+import React, { Fragment, useState, useEffect } from 'react';
+import { NavLink, Route, useHistory } from 'react-router-dom';
 
 import {
-  getSomething
+  getAllProducts,
+  getAllOrders
 } from '../api';
 
-const App = () => {
-  const [message, setMessage] = useState('');
+import { 
+  Product, 
+  AllProducts, 
+  Nav, 
+  Login, 
+  Register, 
+  AllOrders, 
+  SingleOrder, 
+  Cart,
+  Footer, 
+  SingleProduct,
+  SingleUser } from './index';
+
+// import SingleOrder from './SingleOrder';
+import './App.css';
+
+const App = (props) => {
+  const {orderId, userId, cart, setCart, singleProduct, setSingleProduct, singleUser, setSingleUser} = props;
+  const [user, setUser] = useState('');
+  const [token, setToken] = useState('');
+  const [nav, setNav] = useState([]);
+  const [product, setProduct] = useState([]);
+  const [allProducts, setAllProducts] = useState([]);
+  const [singleOrder, setSingleOrder] = useState([]);
+  const [allOrders, setAllOrders] = useState([]);
 
   useEffect(() => {
-    getSomething()
-      .then(response => {
-        setMessage(response.message);
+    const currentUser = localStorage.getItem('user')
+    const currentToken = localStorage.getItem('token')
+    if(currentUser && currentToken){
+      const loggedInUser = JSON.parse(currentUser)
+      const loggedInUserToken = JSON.parse(currentToken)
+      setUser(loggedInUser)
+      setToken(loggedInUserToken)
+    }
+  }, []);
+
+  const fetchProducts = () => {
+    getAllProducts()
+      .then( responseAllProducts => {
+        setAllProducts(responseAllProducts)
+      console.log('responseAllProducts: ', responseAllProducts);
+      })
+  }
+
+  const fetchOrders =() => {
+    getAllOrders()
+      .then( responseAllOrders => {
+        setAllOrders(responseAllOrders)
+      console.log('responseAllOrders: ', responseAllOrders);
       })
-      .catch(error => {
-        setMessage(error.message);
-      });
-  });
-
-  return (
-    <div className="App">
-      <h1>Hello, World!</h1>
-      <h2>{ message }</h2>
+  }
+
+  useEffect(()=>{
+    fetchProducts()
+    fetchOrders()
+  },[])
+
+  return <>
+    <div id="App">
+      <Route>
+        <Nav 
+          token = {token}
+          setToken = {setToken}
+          user = {user}
+          setUser = {setUser}
+        />
+      </Route>
+
+      <div className="welcomeDiv">Welcome to Dope Soap!<br />
+        Enjoy a clean view of all our products!
+      </div>
+     
+      {!token
+      ?
+      <Fragment>
+        <Route path="/Login">
+          <Login
+            setUser = {setUser}
+            setToken = {setToken}
+          />
+        </Route>
+
+        <Route path="/Register">
+          <Register
+            setUser = {setUser}
+            setToken = {setToken}
+          />
+        </Route>
+      </Fragment>
+      :
+     <div>
+      <Fragment>
+        <Route path="/SingleUser">
+          <Register
```

Comment:
Let's create a new component to display a single user?  Just needs a user ID?  OR it can display the user data from the logged in user?  Like getting the data from `/api/users/me`.  Currently this is displaying the register component